src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerChangeService.java

```diff
@@ -322,9 +322,9 @@ private <P extends Officer> boolean setCommonAttributes(
             Boolean addressesAreSameFlag
     ) {
 
-        var hasChange = changeManager.compareAndBuildRightChange(
+        var hasChange = changeManager.compareAndBuildLeftChange(
                 Boolean.TRUE.equals(addressesAreSameFlag) ? residentialAddress : serviceAddress,
-                ManagingOfficerDataApi::getPrincipalAddress,
+                CompanyOfficerApi::getAddress,
```

Comment:
Are you comparing like for lile? Here as CompanyOfficerApi::getPrincipleAddress() is part of the ManagingOfficerApi too. 
But only one address is available for Individual Managing Officers.